File: Behavioral/Lab15/Typescript/src/program.ts

import { AddApartmentCommand } from './Commands/add_apartment_command';
import { AddHouseCommand } from './Commands/add_house_command';
import { AddRoomCommand } from './Commands/add_room_command';
import { RemoveEstateCommand } from './Commands/remove_estate_command';
import { ICommand } from './Interfaces/command';
class Program {
    static main(args: string[]): void {
        console.log("Start testing... \n\n");
        Program.executeCommands([
            new AddApartmentCommand(),
            new AddHouseCommand("house", "house", "1", "1", "test"),
            new AddRoomCommand(),
            new RemoveEstateCommand("apartment", "0"),
            new RemoveEstateCommand("house", "2"),
            new RemoveEstateCommand("house", "1"),
        ]);
        console.log("End testing... \n\n");
    }

    private static executeCommands(commands:ICommand[]): void {
        commands.forEach(command => console.log(command.execute()));
    }
}

Program.main([]);

File: Behavioral/Lab15/C#/Program.cs
using System;
using System.Diagnostics;
using System.Net.WebSockets;
using System.Reflection.Emit;
using System.Runtime.InteropServices;
using C_.Builders;
using C_.TemplateMethods;
using C_.Directors;
using C_.Entities;

namespace C_
{    
    class Program
    {
        static void Main(string[] args)
        {
            Console.WriteLine("Start testing... \n\n");
            ExcecuteCommands(new List<Interfaces.ICommand>{
                new AddApartmentCommand(),
                new AddHouseCommand("house", "house", "1", "1", "test"),
                new AddRoomCommand(),
                new RemoveEstateCommand("apartment", "0"),
                new RemoveEstateCommand("house", "2"),
                new RemoveEstateCommand("house", "1"),
            });
            Console.WriteLine("End testing... \n\n");
        }

        private static void ExcecuteCommands(IEnumerable<Interfaces.ICommand> commands){
            foreach(var command in commands){
                Console.WriteLine(command.Execute());
            }
        }

    }
}
